internal/dag/status_test.go

```diff
@@ -11390,6 +11391,491 @@ func TestGatewayAPITCPRouteDAGStatus(t *testing.T) {
+					},
+					&HTTPProxyProcessor{},
+					&GatewayAPIProcessor{
+						FieldLogger: fixture.NewTestLogger(t),
+					},
+				},
+			}
+
+			// Set a default gateway if not defined by a test
+			if tc.gateway == nil {
+				builder.Source.gateway = &gatewayapi_v1beta1.Gateway{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "contour",
+						Namespace: "projectcontour",
+					},
+					Spec: gatewayapi_v1beta1.GatewaySpec{
+						Listeners: []gatewayapi_v1beta1.Listener{{
+							Name:     "http",
+							Port:     80,
+							Protocol: gatewayapi_v1.HTTPProtocolType,
+							AllowedRoutes: &gatewayapi_v1beta1.AllowedRoutes{
+								Namespaces: &gatewayapi_v1beta1.RouteNamespaces{
+									From: ref.To(gatewayapi_v1.NamespacesFromAll),
+								},
+							},
+						}},
+					},
+				}
+			}
+
+			for _, o := range tc.objs {
+				builder.Source.Insert(o)
+			}
+
+			dag := builder.Build()
+			gotBackendTLSPolicyUpdates := dag.StatusCache.GetBackendTLSPolicyUpdates()
+
+			ops := []cmp.Option{
+				cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+				cmpopts.IgnoreFields(status.BackendTLSPolicyStatusUpdate{}, "GatewayRef"),
+				cmpopts.IgnoreFields(status.BackendTLSPolicyStatusUpdate{}, "Generation"),
+				cmpopts.IgnoreFields(status.BackendTLSPolicyStatusUpdate{}, "TransitionTime"),
+				cmpopts.IgnoreFields(status.BackendTLSPolicyStatusUpdate{}, "Resource"),
+				cmpopts.IgnoreFields(status.GatewayStatusUpdate{}, "ExistingConditions"),
+				cmpopts.IgnoreFields(status.GatewayStatusUpdate{}, "Generation"),
+				cmpopts.IgnoreFields(status.GatewayStatusUpdate{}, "TransitionTime"),
+				cmpopts.SortSlices(func(i, j metav1.Condition) bool {
+					return i.Message < j.Message
+				}),
+				cmpopts.SortSlices(func(i, j *status.BackendTLSPolicyStatusUpdate) bool {
+					return i.FullName.String() < j.FullName.String()
+				}),
+			}
+
+			// Since we're using a single static GatewayClass,
+			// set the expected controller string here for all
+			// test cases.
+			for _, u := range tc.wantBackendTLSPolicyConditions {
+				u.GatewayController = builder.Source.gatewayclass.Spec.ControllerName
+
+				for _, pas := range u.PolicyAncestorStatuses {
+					pas.ControllerName = builder.Source.gatewayclass.Spec.ControllerName
+				}
+			}
+
+			if diff := cmp.Diff(tc.wantBackendTLSPolicyConditions, gotBackendTLSPolicyUpdates, ops...); diff != "" {
+				t.Fatalf("expected backend tls policy status: %v, got %v", tc.wantBackendTLSPolicyConditions, diff)
+			}
+		})
+	}
+
+	tlsService := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "tlssvc",
+			Namespace: "projectcontour",
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{makeServicePort("https", "TCP", 443, 8443)},
+		},
+	}
+
+	configMapCert1 := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ca",
+			Namespace: "projectcontour",
+		},
+		Data: map[string]string{
+			CACertificateKey: fixture.CERTIFICATE,
+		},
+	}
+
```

Comment:
Another would be to have multiple CACertRefs, I think the optimal case would be with 3, 2 invalid, one valid so we get all the current uncovered code

This should exercise the logic we have in `BackendTLSPolicyAncestorStatusUpdate.AddCondition` since we would be adding the same condition type multiple times. It would also mean we might need to take a look at the logic here: https://github.com/projectcontour/contour/pull/6151/files#diff-2dfb3932d84ad36571b20cb8d66d9f0b623fdbbef49a0b41eff48eaa7fa22a05R2150

Reply:
e.g. a test with this set of ca certs

```
						CACertRefs: []gatewayapi_v1alpha2.LocalObjectReference{
							{
								Kind: "Invalid",
								Name: gatewayapi_v1.ObjectName(configMapCert1.Name),
							},
							{
								Kind: "ConfigMap",
								Name: gatewayapi_v1.ObjectName("missing"),
							},
							{
								Kind: "ConfigMap",
								Name: gatewayapi_v1.ObjectName(configMapCert1.Name),
							},
						},
```

Reply:
To be conservative we can maybe say `Accepted: false` for Policies that have anything invalid (right now a Policy with this config will be `Accepted: true` since it has one valid cert ref)

Reply:
Added both suggested test cases.

Good catch on the mix of valid and invalid certs. Added a guard to catch if any certs in the chain are invalid and marking the backendtlspolicy as `Accepted: false`.